sympy/polys/densearith.py

```diff
@@ -997,6 +1031,46 @@ def dmp_pow(f, n, u, K):
     return g
 
 
+def _dup_recurse_pow(f, exp, prec, K):
+    """
+    Helper function for dup_series_pow.
+    """
+    if exp == 1:
+        return dup_truncate(f, prec, K)
+
+    q, r = divmod(exp, 2)
+    half = _dup_recurse_pow(f, q, prec, K)
+    square = dup_series_mul(half, half, prec, K)
+
+    if r == 0:
+        return square
+    return dup_series_mul(square, f, prec, K)
+
+
+def dup_series_pow(f, n, prec, K):
+    """
+    Raise ``f`` to the ``n``-th power in ``K[[x]]`` up to ``x**prec``.
```

Comment:
It is not completely clear what `x**prec` means e.g. does it mean:
```
a + b*x + ... + c*x**prec + O(x**(prec + 1))
```
I think it is better to say `mod x**prec` or something,